p2p/protocol/identify/id.go

```diff
@@ -344,18 +394,35 @@ func (ids *IDService) populateMessage(mes *pb.Identify, c network.Conn) {
 	// "public" address, at least in relation to us.
 	mes.ObservedAddr = c.RemoteMultiaddr().Bytes()
 
-	// set listen addrs, get our latest addrs from Host.
-	laddrs := ids.Host.Addrs()
-	// Note: LocalMultiaddr is sometimes 0.0.0.0
-	viaLoopback := manet.IsIPLoopback(c.LocalMultiaddr()) || manet.IsIPLoopback(c.RemoteMultiaddr())
-	mes.ListenAddrs = make([][]byte, 0, len(laddrs))
-	for _, addr := range laddrs {
-		if !viaLoopback && manet.IsIPLoopback(addr) {
-			continue
+	if usePeerRecords {
+		ids.peerrecMu.RLock()
+		rec := ids.peerrec
+		ids.peerrecMu.RUnlock()
+
+		if rec == nil {
+			log.Errorf("latest peer record does not exist. identify message incomplete!")
+		} else {
+			recBytes, err := rec.Marshal()
+			if err != nil {
+				log.Errorf("error marshaling peer record: %v", err)
+			} else {
+				mes.SignedPeerRecord = recBytes
+				log.Debugf("%s sent peer record to %s", c.LocalPeer(), c.RemotePeer())
+			}
+		}
+	} else {
+		// set listen addrs, get our latest addrs from Host.
+		laddrs := ids.Host.Addrs()
+		// Note: LocalMultiaddr is sometimes 0.0.0.0
+		viaLoopback := manet.IsIPLoopback(c.LocalMultiaddr()) || manet.IsIPLoopback(c.RemoteMultiaddr())
```

Comment:
@Stebalien 

Hmm.. we probably want to imitate this logic for the signed addresses ?

Reply:
it's probably also worth filtering `manet.IsIPUnspecified()` records in both cases.

Reply:
@Stebalien Could you point me to an issue/discussion/PR to better understand the idea and scope of the address scoping work ?

Reply:
@aarshkshah1992 I would imagine the intention would be to split record/addresses along the same policy distinction as in https://github.com/libp2p/go-libp2p-kad-dht/blob/master/dht_filters.go - a public scope for addresses the peer thinks might be valid for remote peers and is willing to drop on the public DHT, and separately the set of addresses valid for LAN peers.

Reply:
https://github.com/libp2p/go-libp2p/issues/793